src/xbot/common/command/scripted/ScriptedCommand.java

```diff
@@ -0,0 +1,164 @@
+package xbot.common.command.scripted;
+
+import java.io.File;
+import java.lang.Thread.State;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.apache.log4j.Priority;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Scheduler;
+
+/**
+ * A command to facilitate execution of a JavaScript script with access
+ * to robot functionality such as Command invocation.
+ *
+ */
+public class ScriptedCommand extends Command {
+
+    static Logger log = Logger.getLogger(ScriptedCommand.class);
+    
+    private ScriptedCommandThread execThread;
+
+    ScriptedCommandFactory availableCommandFactory;
+    
+    File scriptFile = null;
+    String manualScriptText = null, manualScriptName = null;
+    
+    /**
+     * Contains all the commands that this script has invoked.
+     * NOTE: Not all commands in this collection are guaranteed to be running;
+     * being in this list only signifies that a command was invoked at some
+     * point by this script.
+     */
+    private Set<Command> invokedCommands;
+    
+    private volatile Set<ScriptedCommandCompletionLock> completionLocks;
+    
+    public ScriptedCommand(File scriptFile, ScriptedCommandFactory availableCommandFactory) {
+        this.availableCommandFactory = availableCommandFactory;
+        
+        this.scriptFile = scriptFile;
+        
+        log.debug("Instantiated with file path\"" + scriptFile.getAbsolutePath() + "\"");
+    }
+    
+    public ScriptedCommand(String manualScriptText, String manualScriptName, ScriptedCommandFactory availableCommandFactory) {
+        this.availableCommandFactory = availableCommandFactory;
+        
+        this.manualScriptText = manualScriptText;
+        this.manualScriptName = manualScriptName;
+        
+        log.debug("Instantiated with script string");
+    }
+    
+    private void initializeExecThread() {
+        log.debug("Initializing exec thread");
```

Comment:
It looks like this is only invoked once - might be a good candidate for info logging.